### PR TITLE
Add pending layer changes view

### PR DESCRIPTION
**Pull request by NathanW2**
Super early preview of feature for feedback and ideas

image

### CODE REVIEW

**Comment by m-kuhn** [comment]
@NathanW2 can we leave 3.4 as milestone here? Looks like a superinteresting feature but also like one you might have forgotten about ;)

**Comment by NathanW2** [comment]
Yeah we can leave it if you want. I would like to come back to this at some
stage but yes you are right I just forgot about it.

**Comment by NathanW2** [comment]
@m-kuhn Yeah I'm keen to revist this for 3.4 just need to flesh it out a bit more maybe   Was really only a prototype here.

File: src/gui/pendingchanges/qgspendingchangesmodel.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+#include <QStandardItem>
+
+QgsPendingChangesModel::QgsPendingChangesModel( QgsProject *project, QObject *parent )
+  : QStandardItemModel( parent )
+{
+  setColumnCount( 2 );
+  setHorizontalHeaderLabels( QStringList() << "Item" << "Layer" );
+  mAddedFeaturesRoot = new QStandardItem( tr( "Added features" ) );
+  mChagnedFeaturesRoot = new QStandardItem( tr( "Modified features" ) );
```
Comment (elpaso) [inline src/gui/pendingchanges/qgspendingchangesmodel.cpp]: typo: chagned

File: src/gui/pendingchanges/qgspendingchangesmodel.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      // TODO Swap this for featuresDeleted for better speed
+      connect( vLayer, &QgsVectorLayer::featureDeleted, this, [this, vLayer]( QgsFeatureId id )
+      {
+        featureDeleted( id, vLayer );
+      } );
+      connect( vLayer, &QgsVectorLayer::attributeValueChanged, this, [this, vLayer]( QgsFeatureId id )
+      {
+        featureModified( id, vLayer );
+      } );
+
```
Comment (elpaso) [inline src/gui/pendingchanges/qgspendingchangesmodel.cpp]: What about geometry changed?

**Comment by elpaso** [comment]
@NathanW2 thanks! very much welcome feature!

**Pull request closed without merging**